src/common/post.ts

```diff
@@ -329,3 +344,125 @@ export const insertCodeSnippetsFromUrl = async ({
     throw err;
   }
 };
+
+type ModeratedPostCdc = ChangeObject<SourcePostModeration>;
+
+export const updateModeratedPost = async (
+  con: ConnectionManager,
+  moderated: ModeratedPostCdc,
+): Promise<ModeratedPostCdc | null> => {
+  if (!moderated?.postId) {
+    logger.error('unable to update moderated post', { moderated });
+    return null;
+  }
+
+  const postParam = { id: moderated.postId };
+  const repo = con.getRepository(Post);
+
+  await repo.findOneOrFail({ where: { id: moderated.postId } });
+
+  const updatedPost: Partial<FreeformPost> = {
+    title: moderated.title,
+    titleHtml: moderated.titleHtml,
+  };
+
+  if (moderated.content) {
+    updatedPost.content = moderated.content;
+    updatedPost.contentHtml = moderated.contentHtml!;
```

Comment:
Don't we do stuff with markdown or tags here? (can't recall so top of mind)

Reply:
Yes, and since from the post submission we generate it already. Hence, we just pass it here.

Reply:
Ah ok, guess just very hard to follow from review which part actually does something 😢 
One day we need to clean this up a bit I feel.

Reply:
Yeah, totally agree. Feels like we need some sort of process flow that will handle everything when it comes to creating post.

Reply:
Let's log it for backlog, don't want to keep this project blocked by it.

Reply:
@sshanzel 

> Totally agree with that, although I tried to make them generic as much as I can, but we have three different processes for 3 different kind of submissions. For Freeform, we just create everything the user adds, for share, we have two kinds. One is sharing already existing post, the other accepts external link. All 3 of them have different requirements, for example, Share does not need content, then on external link we have a title but going to be used on the article post generated then generate another share post with the content.

This is a very good point, ideally this is just wrapped in 3 functions then:
- `createFreeformPost`
- `createShareExistingPost`
- `createShareExternalLinkPost`

And those rule all the things, but it's more tech debt clean up as this point.

Reply:
Yeah, that would be best. We actually apply that now here, hopefully, we can apply it on the legacy code too. 

Ticket created: https://dailydotdev.atlassian.net/jira/software/projects/WT/boards/11/backlog?selectedIssue=WT-2245